aten/src/ATen/native/TensorShape.cpp

```diff
@@ -394,27 +394,34 @@ Tensor& stack_out(Tensor& result, TensorList tensors, int64_t dim) {
 }
 
 static inline Tensor & sparse_transpose_(Tensor & self, int64_t dim0, int64_t dim1) {
-  int64_t ndimI = self._indices().size(0);
+  int64_t ndimI = self._dimI();
   if (dim0 >= ndimI || dim1 >= ndimI) {
     AT_ERROR(
-        "sparse transpose_: transposed dimensions must be sparse ",
+        "sparse transpose: transposed dimensions must be sparse ",
         "Got nDimI: ", ndimI, ", d0: ", dim0, ", d1: ", dim1);
   }
 
-  auto indices = self._indices();
-  auto row0 = indices.select(0, dim0);
-  auto row1 = indices.select(0, dim1);
+  if (self._indices().numel() == 0 && self._values().numel() == 0) {
+    std::vector<int64_t> sizes(self.sizes());
+    std::swap(sizes[dim0], sizes[dim1]);
 
-  // swap row0 and row1
-  auto tmp = at::zeros_like(row0);
-  tmp.copy_(row0);
-  row0.copy_(row1);
-  row1.copy_(tmp);
+    return self.sparse_raw_resize_(sizes, self._dimI(), self._dimV());
+  } else {
```

Comment:
nit: don't need the else block (there's a return in the if block)

Reply:
IMO it's easier to follow with the else (you don't have to understand anything in the if.